includes/admin/class-wcs-admin-post-types.php

```diff
@@ -249,7 +249,7 @@ public function print_bulk_actions_script() {
 
 		$post_status = ( isset( $_GET['post_status'] ) ) ? $_GET['post_status'] : '';
 
-		if ( 'shop_subscription' !== get_post_type() || in_array( $post_status, array( 'cancelled', 'trash', 'wc-expired' ) ) ) {
+		if ( 'shop_subscription' !== WC_Data_Store::load( 'subscription' )->get_order_type() || in_array( $post_status, array( 'cancelled', 'trash', 'wc-expired' ), true ) ) {
```

Comment:
`til:` if no id is passed, it will use `$GLOBALS['post']` ([ref](https://github.com/WordPress/WordPress/blob/25ea9680aecaf4791c870658fd8085c10622ad05/wp-includes/post.php#L991-L993)).

Reply:
Thanks for pointing this out @shendy-a8c. This complicates things a little because if this function is called on stores that have HPOS enabled, the global post won't exist and the [OrdersTable data store class](https://github.com/woocommerce/woocommerce/blob/a734772c69f953fa466e861670f39732f0c83c7f/plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php#L949) must be sent with an ID otherwise an error will be thrown.

Even though the postID is defaulted in `get_post_type()` when the CPT data store is loaded, we'll need to make sure this function doesn't break when the HPOS data store is loaded.

That means we can either:

1. Return early if store has HPOS enabled.
2. Look for and fetch an ID from `$GLOBALS['post']` if it exists. If no post/ID is found, then we return early from the function.

Personally I like option 2 because it's a bit more defensive, but I'm happy with either option :)

Reply:
That makes sense. Addressed in [92c767f](https://github.com/Automattic/woocommerce-subscriptions-core/pull/318/commits/92c767fbcc7ac2ade5efd2f4f19404a36085e5ab).